cmd/var.go

```diff
@@ -0,0 +1,126 @@
+package cmd
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/driusan/dgit/git"
+)
+
+func getLogVar(c *git.Client, config *git.GitConfig, key string, envVar string) string {
+	if envVar != "" {
+		return envVar
+	}
+
+	if key == "GIT_AUTHOR_IDENT" {
+		name := os.Getenv("GIT_AUTHOR_NAME")
+		email := os.Getenv("GIT_AUTHOR_EMAIL")
+		date := os.Getenv("GIT_AUTHOR_DATE")
+		if name != "" && email != "" && date != "" {
+			return fmt.Sprintf("%s <%s> %s", name, email, date)
+		}
+	}
+
+	if key == "GIT_COMMITTER_IDENT" {
+		name := os.Getenv("GIT_COMMITTER_NAME")
+		email := os.Getenv("GIT_COMMITTER_EMAIL")
+		date := os.Getenv("GIT_COMMITTER_DATE")
+		if name != "" && email != "" && date != "" {
+			return fmt.Sprintf("%s <%s> %s", name, email, date)
+		}
+	}
+
+	if key == "GIT_AUTHOR_IDENT" || key == "GIT_COMMITTER_IDENT" {
+		t := time.Now()
+		person := c.GetAuthor(&t)
+		return fmt.Sprintf("%s <%s> %v %s", person.Name, person.Email, time.Now().Unix(), time.Now().Format("-0700"))
+	} else if key == "GIT_EDITOR" {
+		coreEditor, _ := config.GetConfig("core.editor")
+		if coreEditor != "" {
+			return coreEditor
+		}
+
+		visual := os.Getenv("VISUAL")
```

Comment:
I think this is missing a check for os.Getenv("GIT_EDITOR") (I'd have to look up the precedence, but I assume before the config)

Reply:
It's handled as a catch-all for all vars at the top of the function. The environment variable value that matches the key is passed in to this function.

Reply:
Ah, I missed that